init.lua

```diff
@@ -1173,6 +1173,12 @@ function gui.Spacer(def)
     return def
 end
 
+-- for use in inline <bool> and <a> or <b> type inline ifs
+function gui.Nil(def)
+    def.expand = false
+    return gui.Spacer(def)
```

Comment:
Using `gui.Spacer{expand = false}` is slightly different from having no element at all since spacing is still added around the spacer (<a href="https://luk3yx.gitlab.io/minetest-flow-playground/#code=local%20gui%20%3D%20flow.widgets%0A%0Alocal%20my_gui%20%3D%20flow.make_gui(function(player%2C%20ctx)%0A%20%20%20%20return%20gui.HBox%7B%0A%20%20%20%20%20%20%20%20gui.VBox%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Label%7Blabel%20%3D%20%22No%20element%22%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Label%7Blabel%20%3D%20%22Second%20label%22%7D%2C%0A%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%20%20gui.VBox%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Label%7Blabel%20%3D%20%22Spacer%20with%20expand%3Dfalse%22%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Spacer%7Bexpand%20%3D%20false%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Label%7Blabel%20%3D%20%22Second%20label%22%7D%2C%0A%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%20%20gui.VBox%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Label%7Blabel%20%3D%20%22Tooltip%22%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Tooltip%7Bvisible%20%3D%20false%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20gui.Label%7Blabel%20%3D%20%22Second%20label%22%7D%2C%0A%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%7D%0Aend)%0A%0Amy_gui%3Ashow(minetest.get_player_by_name(%22playground%22))%0A">playground link</a>):

![image](https://user-images.githubusercontent.com/3182651/232904197-c94aa845-43ab-4ca3-93a2-167f0347bf1a.png)

Maybe it'd be better to use `tooltip` with `visible = false`, there's a special case for some elements like tooltip and style so that they aren't layouted.

Reply:
Interesting. This could explain some odd behavior I've been encountering.